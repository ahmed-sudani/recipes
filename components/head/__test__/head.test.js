import { render, screen } from '@testing-library/react'
import Head from '..'

jest.mock('next/head', () => {
  return {
    __esModule: true,
    default: ({ children }) => {
      return <div data-testid="head">{children}</div>
    },
  }
})

describe('Head', () => {
  it('should render correctly', async () => {
    render(<Head title={'test'} description="this is test" />)
    const headElement = screen.getByTestId(/head/i)
    expect(headElement).toMatchSnapshot()
  })
  it('title should match the text passed', async () => {
    render(<Head title={'test'} description="this is test" />)
    expect(document.title).toEqual('test')
  })
})
