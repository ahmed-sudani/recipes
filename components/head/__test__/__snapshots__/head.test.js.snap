// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head should render correctly 1`] = `
<div
  data-testid="head"
>
  <title
    data-testid="test"
  >
    test
  </title>
  <meta
    content="this is test"
    data-testid="this is test"
    name="description"
  />
  <link
    href="/favicon.ico"
    rel="icon"
  />
</div>
`;
